docs/source/config_files.rst

```diff
@@ -164,6 +164,27 @@ used for installing libraries.
 To see an example R repository, visit our `R
 example in binder-examples <https://github.com/binder-examples/r/blob/master/runtime.txt>`_.
 
+.. _default.nix:
+
+``default.nix`` - the nix package manager
+=========================================
+
+Specify packages to be installed by the `nix package manager <https://github.com/NixOS/nixpkgs>`_.
+When you use this config file all other configuration files (like ``requirements.txt``)
+that specify packages are ignored. When using ``nix`` you have to specify all
+packages and dependencies explicitly, including the Jupyter notebook package that
+repo2docker expects to be installed. If you do not install Jupyter explicitly
+repo2docker will no be able to start your container.
+
+`nix-shell <https://nixos.org/nix/manual/#sec-nix-shell>`_ is used to evaluate
+a ``nix`` expression written in a ``default.nix`` file. Make sure to
+`pin your nixpkgs <https://discourse.nixos.org/t/nixops-pinning-nixpkgs/734>`_
+to produce a reproducible environment.
+
+To see an example repository visit
+`nix binder example <https://gitlab.com/costrouc/nix-binder-example>`_.
```

Comment:
One last thing that is my fault. Could you change the link to github.com/costrouc/nix-binder-example

Otherwise everything looks great to me! Thank you for your work on making this compatible.

Reply:
I'll say that getting `nix` to work in the build pack the first time was not easy. Especially the part `nix-shell default.nix --command ....`. Yeah I would be happy to transfer over the repo. I made two due to https://github.com/costrouc/nix-binder-example/issues/1.

Would you like [nix-binder-example](https://github.com/costrouc/nix-binder-example) or [nix-binder-example-simple](https://github.com/costrouc/nix-binder-example-simple)?

Reply:
Let's start with the simple one?

Reply:
If you accept the invitation to the binder-examples org you should be able to transfer the repo.

Reply:
Done! Thanks.The repository will most likely need a rename. As far as I can tell I don't have permissions to do that.

Reply:
Also this link will need to be updated. Looks like you have commit access so it would be an easy change once you set the name. 

https://github.com/jupyterhub/binder/blob/master/doc/examples.rst#scientific-research-and-reproducibility